### ISSUE TITLE
LoRA for Dreambooth

### ISSUE THREAD

**Issue by polavishnu4444** [huggingface/diffusers#1715]
Hi,
Have come across LoRA for diffusion models. 
It might be right for diffusers to collaborate and later get to one single diffusers SDK as Dreambooth is supported here as well in the current diffusers' library.
Interested to learn about your thoughts on this. 😃 

Link: https://github.com/cloneofsimo/lora

**Comment by patrickvonplaten** [comment]
Yes, very interested in adding this to `diffusers`. Would love to review a PR here. Note that it could be as simple to copy the correct training script from: https://huggingface.co/spaces/ysharma/Low-rank-Adaptation/tree/main

**Comment by patrickvonplaten** [comment]
Also cc @pcuenca @williamberman @patil-suraj

**Comment by patil-suraj** [comment]
LMK if anyone is interested in adding this, very happy to help :)

**Comment by Thomas-MMJ** [comment]
@cloneofsimo

**Comment by cloneofsimo** [comment]
Ah sorry we've actually talked about this on the slack channel as well. Just finished final exam. I'll work on it today

**Comment by dhruvrnaik** [comment]
Happy to help with it @cloneofsimo

**Comment by polavishnu4444** [comment]
Interested as well if you can teach me 😃

**Comment by shirayu** [comment]
Just for your information: 

This repository includes another LoRA implementation, however I haven't tried.
https://github.com/kohya-ss/sd-scripts

Description in Japanese:
https://github.com/kohya-ss/sd-scripts/blob/main/train_network_README-ja.md

They stated that the main difference between the cloneofsimo's implementation at this time is more application of LoRA to MLP in Text Encoder, FFN in U-Net, and in/out projection in Transformer.

**Comment by patrickvonplaten** [comment]
Starting working on it: https://github.com/huggingface/diffusers/pull/1884 -> should be ready end of week :-)

**Comment by cloneofsimo** [comment]
I'm sorry for not updating this earlier. And thank you @patrickvonplaten !  I just checked this PR as well : https://github.com/huggingface/diffusers/pull/1639 this would make LoRA and Paint with words implementation much easier. (But would probably need refactor as well) 

I've been trying to make LoRA better before implementing here, and with contributors, we've recently made 4 major progresses and improved fidelity considerably more. I'll try my best to implement them in the examples when I find time (or anyone can jump in)
